include/ppx/string_util.h

```diff
@@ -31,10 +35,18 @@ std::string TrimCopy(const std::string& s);
 // Trims all characters specified in c from both the left and right sides of s
 std::string_view TrimBothEnds(std::string_view s, std::string_view c = " \t");
 
+// Splits s at every instance of delimeter and returns a vector of substrings
+// Returns std::nullopt if s contains: leading/trailing/consecutive delimiters
+std::optional<std::vector<std::string_view>> Split(std::string_view s, char delimiter);
```

Comment:
Personally, I think it would be cleaner for these two to just return the value (std::vector or std::pair) without the optional, and for the caller to know and deal with the empty values, etc. But up to you!

Reply:
yes. Also, this is a basic string util, so I'd expect it's behavior to be straightforward:
split at the separator, and return the values in between.
No delimiter -> return the whole string
Multiple delimiters -> returns en empty tokens between each delimiter.